src/main/java/ch/njol/skript/lang/SyntaxStringBuilder.java

```diff
@@ -0,0 +1,117 @@
+package ch.njol.skript.lang;
+
+import com.google.common.base.Preconditions;
+import org.bukkit.event.Event;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.StringJoiner;
+
+/**
+ * Utility class to build syntax strings, used in {@link Debuggable#toString(Event, boolean)}.
+ */
+public class SyntaxStringBuilder {
+
+	private final boolean debug;
+	private final @Nullable Event event;
+	private final StringJoiner joiner = new StringJoiner(" ");
+
+	/**
+	 * Creates a new SyntaxStringBuilder.
+	 *
+	 * @param event The event to get information from. This is always null if debug == false.
+	 * @param debug If true this should print more information, if false this should print what is shown to the end user
+	 */
+	public SyntaxStringBuilder(@Nullable Event event, boolean debug) {
+		this.event = event;
+		this.debug = debug;
+	}
+
+	/**
+	 * Adds a boolean to the string.
+	 * @param b The boolean to add.
+	 * @return This SyntaxStringBuilder.
+	 */
+	public SyntaxStringBuilder add(boolean b) {
```

Comment:
i'd like to see it work like stringbuilder, with `append`, since that's the name you've used for the class.